pkg/runtime/function/cast_decimal.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package function
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+)
+
+import (
+	gxbig "github.com/dubbogo/gost/math/big"
+
+	"github.com/pkg/errors"
+)
+
+import (
+	"github.com/arana-db/arana/pkg/proto"
+	"github.com/arana-db/arana/pkg/util/math"
+)
+
+// FuncCastDecimal is https://dev.mysql.com/doc/refman/5.6/en/cast-functions.html#function_cast
+const FuncCastDecimal = "CAST_DECIMAL"
+
+var _ proto.Func = (*castDecimalFunc)(nil)
+
+func init() {
+	proto.RegisterFunc(FuncCastDecimal, castDecimalFunc{})
+}
+
+type castDecimalFunc struct{}
+
+func (a castDecimalFunc) Apply(ctx context.Context, inputs ...proto.Valuer) (proto.Value, error) {
+	if len(inputs) != 3 {
+		return "", errors.New("The Decimal function must accept three parameters\n")
+	}
+
+	val, err := inputs[0].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	v := math.ToDecimal(val)
+
+	maxNum, err := inputs[1].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	if maxNum == nil {
+		maxNum = 0
```

Comment:
maybe the default max num is 10, refer to https://dev.mysql.com/doc/refman/5.6/en/fixed-point-types.html.